examples/prom/nonlinear_elasticity_global_rom.cpp

```diff
@@ -0,0 +1,1747 @@
+
+            if (def_ic)
+            {
+                deform = new VectorFunctionCoefficient(dim, InitialDeformationIC2);
+            }
+            else
+            {
+                deform = new VectorFunctionCoefficient(dim, InitialDeformationIC1);
+            }
+            sp_x_gf.ProjectCoefficient(*deform);
+            sp_x_gf.SetTrueVector();
+
+            sp_v_gf.SetFromTrueVector();
+            sp_x_gf.SetFromTrueVector();
+
+            // Get initial conditions
+            w_v0 = new Vector(sp_v_gf.GetTrueVector());
+            w_x0 = new Vector(sp_x_gf.GetTrueVector());
+
+            // Convert essential boundary list from FOM mesh to sample mesh
+            // Create binary list where 1 means essential boundary element, 0 means nonessential.
+            CAROM::Matrix Ess_mat(true_size, 1, true);
+            for (size_t i = 0; i < true_size; i++)
+            {
+                Ess_mat(i,0) = 0;
+                for (size_t j = 0; j < ess_tdof_list.Size(); j++)
+                {
+                    if (ess_tdof_list[j] == i )
+                    {
+                        Ess_mat(i,0) = 1;
+                    }
+                }
+            }
+
+            // Project binary FOM list onto sampling space
+            CAROM::Matrix Ess_mat_sp(sp_size, 1, false);
+            smm->GatherDistributedMatrixRows("X", Ess_mat, 1, Ess_mat_sp);
+
+            // Count number of true elements in new matrix
+            int num_ess_sp = 0;
+
+            for (size_t i = 0; i < sp_size; i++)
+            {
+                if (Ess_mat_sp(i,0) == 1)
+                {
+                    num_ess_sp += 1;
+                }
+
+            }
+
+            // Initialize essential dof list in sampling space
+            Array<int> ess_tdof_list_sp(num_ess_sp);
+
+            // Add indices to list
+            int ctr = 0;
+            for (size_t i = 0; i < sp_size; i++)
+            {
+                if (Ess_mat_sp(i,0) == 1)
+                {
+                    ess_tdof_list_sp[ctr] = i;
+                    ctr += 1;
+                }
+
+            }
+
+            // Define operator in sample space
+            soper = new HyperelasticOperator(*sp_XV_space, ess_tdof_list_sp, visc, mu, K);
+
+        }
+
+        if (hyperreduce)
+        {   romop = new RomOperator(&oper, soper, rvdim, rxdim, hdim, smm, *w_v0, *w_x0,
+                                    vx0.GetBlock(0), BV_librom, BX_librom, H_librom, Hsinv, myid,
+                                    num_samples_req != -1, hyperreduce, x_base_only);
+        }
+        else
+        {
+            romop = new RomOperator(&oper, soper, rvdim, rxdim, hdim, smm, vx0.GetBlock(0),
+                                    vx0.GetBlock(1), vx0.GetBlock(0), BV_librom, BX_librom, H_librom, Hsinv, myid,
+                                    num_samples_req != -1, hyperreduce, x_base_only);
+        }
+
+        // Display lifted initial energies
```

Comment:
Would `Print` be more accurate than `Display`? `Display` sounds like a visualization is being done.

Reply:
I agree.